### DISCUSSION TITLE
AuthorizeResource method for PendingResourceRegistration

### DISCUSSION THREAD

**Discussion by JesseKoper** [laravel/framework#43964] [Ideas]
Currently the only two ways to authorize a resource controller are to either add:

```php
$this->authorizeResource(User::class);
```

In the constructor of the controller, or:

```php
$this->authorize('view', $user);
```

In the specific methods.

The only way to do this in the route files is to write all your routes separately and use the `->can()` method for example. As far as I am aware at least.

I've been experimenting with an `->authorizeResource()` option on the resource directly, that functions the same way as the constructor option. For example:

```php
Route::resource('users', UserController::class)->authorizeResource(User::class);
```

This seems to work quite nicely, and I'm wondering if this is something I should try and make a PR for, or if this is more suitable as a package.

**Comment by slavatarsyurimi** [comment]
Looks like a handy idea! Anyone else would like to see a PR for this?
